crates/core/tedge_api/src/workflow/mod.rs

```diff
@@ -113,6 +113,52 @@ impl Display for OperationAction {
 }
 
 impl OperationWorkflow {
+    /// Return a new OperationWorkflow unless there are errors
+    /// such as missing or ill-defined states.
+    pub fn try_new(
+        operation: OperationType,
+        handlers: DefaultHandlers,
+        mut states: HashMap<StateName, OperationAction>,
+    ) -> Result<Self, WorkflowDefinitionError> {
+        // The init state is required
+        if states.get("init").is_none() {
+            return Err(WorkflowDefinitionError::MissingState {
+                state: "init".to_string(),
+            });
+        }
+
+        // The successful state can be omitted,
+        // but must be associated to a `clear` if provided.
+        let action_on_success = states
+            .entry("successful".to_string())
+            .or_insert(OperationAction::Clear);
```

Comment:
`OperationAction::Clear` might be a misnomer. The agent clear nothing but wait for the requester to clear the command topic.

Reply:
Aah okay. My bad. I thought that `Clear` action was giving the ability to automatically clear the operation, as that's what I was trying to do in #2484.

And yes, the naming could be more explicit to avoid such confusion. But, since it's not part of the external workflow TOML API like `action=clear`, it's okay.

Reply:
I just noticed that `action=cleanup` maps to the same `OperationAction::Clear`. That's also a bit misleading IMO, giving the notion that we perform some cleanup. But we can improve these things later.